### ISSUE TITLE
Collector definition not processing "groupid" definitions

### ISSUE THREAD

**Issue by jk1066** [gridlab-d/gridlab-d#1438]
Initial Checklist prior to submitting the issue. Check the bellow boxes to the best of your knowledge:

- [x ] I have not found an open issue describing the same problem.
- [ ] I went through "CONTRIBUTING.MD" before deciding to submit the issue.
- [x ] While building GridLAB-D, I followed all steps as detailed in GridLAB-D Wiki.

------------------------------------------------------------------------------------------------------------

## Please describe what was the cause of the issue?

The collector object doesn't recognize groupid definitions, saying it doesn't result in a set with a fixed class

## Please provide the steps to reproduce.

object collector {
    group "groupid=playerhouses";
    name coolON_collector;
    file "avgcoolON.csv";
    interval 60;
    property avg(is_COOL_on);
}
OR
object collector {
    group "groupid=playerhouses AND class=house";
    name coolON_collector;
    file "avgcoolON.csv";
    interval 60;
    property avg(is_COOL_on);
}

## Please provide the exact error/warning/failure message printed on the console.

The first
ERROR    [2022-07-16 00:00:00 CDT] : aggregate group expression 'groupid=playerhouses' does not result in a set with a fixed class
ERROR    [2022-07-16 00:00:00 CDT] : collector 12267 'avg(air_temperature)' contains an aggregate that is not found in the group 'groupid=playerhouses'
ERROR    [2022-07-16 00:00:00 CDT] : 2022-07-16 00:00:00 CDT: object air_temp_collector stopped its clock (exec)!

The second: 
ERROR    [2022-07-16 00:00:00 CDT] : aggregate group expression 'groupid=playerhouses AND class=house' does not result in a set with a fixed class
ERROR    [2022-07-16 00:00:00 CDT] : collector 12267 'avg(air_temperature)' contains an aggregate that is not found in the group 'groupid=playerhouses'
ERROR    [2022-07-16 00:00:00 CDT] : 2022-07-16 00:00:00 CDT: object air_temp_collector stopped its clock (exec)!

## If possible, please provide any screenshots and other supporting information.

If you rearrange the definition so that class is first, it will run without error, but it will ignore the groupid definition. 

object collector {
    group "class=house AND groupid=playerhouses ";
    name coolON_collector;
    file "avgcoolON.csv";
    interval 60;
    property avg(is_COOL_on);
}

Works, but will collect for all class house, not restricting to the ones with the groupid=playerhouses
-------------------------------------------------------------------------------------------------------------

**Comment by ftuffner** [comment]
Of note is `collector` and `group_recorder` seem to handle the `group` field differently.  Simple test file attached for trying the two, but `collector` may need to be harmonized with how `group_recorder` is doing it (to be consistent/proper).
[test_collector_group_recorder.glm.txt](https://github.com/gridlab-d/gridlab-d/files/11637900/test_collector_group_recorder.glm.txt)

Also not sure the "AND" construct is being properly parsed by `collector` objects, likely due to the difference noted above.

**Comment by afisher1** [comment]
yes Frank I don't think collector was written with the same fidelity as the group_recorder.
